Snakefile

```diff
@@ -214,14 +214,22 @@ def collect_prepared_input(wildcards):
     prepared_inputs = expand(f'{prepared_dir}{SEP}{{type}}.txt',type=runner.get_required_inputs(algorithm=wildcards.algorithm))
     # If the directory is missing, do nothing because the missing output triggers running prepare_input
     if os.path.isdir(prepared_dir):
-        # If the directory exists, confirm all prepared input files exist as well (as opposed to some or none)
-        missing_inputs = False
-        for input in prepared_inputs:
-            if not os.path.isfile(input):
-                missing_inputs = True
-        # If any expected files were missing, delete the entire directory so the call below triggers running prepare_input
-        if missing_inputs:
-            shutil.rmtree(prepared_dir)
+        # First, check if .snakemake_timestamp, the last written file in a directory rule,
+        # exists. This prevents multithreading errors if we accidentally read a directory
+        # before it finishes. A proper API for querying this is opened as an issue at
+        # https://github.com/snakemake/snakemake/issues/439.
```

Comment:
The worst-case scenario here is that someone runs two Snakemake files with differing `generate_input` functions at the same time. I also can not imagine a realistic scenario where this happens other than on some strangely shared-fs dev machine.

Reply:
Note: This happens because this entire rule isn't sequential. That's the heart of this problem: the wildcards are being parallelized across threads, despite multiple parameters locking in to a single folder. The other solution to fix this is to attach the parameters hash to the prepare-inputs folder, but that is a lot of unnecessary copying.

Reply:
Was the same work happening on multiple threads or just the same wildcards were being sent to multiple threads?